src/files/BrsFile.Class.spec.ts

```diff
@@ -1178,4 +1186,340 @@ describe('BrsFile BrighterScript classes', () => {
             file.parser.references.classStatements[0].getParentClassIndex(new BrsTranspileState(file));
         });
     });
+
+    describe('getHover', () => {
+
+        const animalClassCode = `
+        class Animal
+            kind as string
+            isHungry as boolean
+
+            sub new(kind as string)
+                m.kind = kind
+                m.isHungry = true ' born hungry
+            end sub
+
+            sub eat(foodAmount as integer)
+                if foodAmount > 100
+                  m.isHungry = false
+                end if
+            end sub
+
+            sub sleep()
+              m.isHungry = false
+            end sub
+
+        end class
+
+        class Dog extends Animal
+            breed as string
+            sub new(breed as string)
+                super("Dog")
+                m.breed = breed
+            end sub
+
+            sub snooze()
+                m.sleep()
+            end sub
+        end class
+
+        class DogHouse
+            puppy as Dog
+            sub new(pup as Dog)
+                m.puppy = pup
+            end sub
+        end class
+
+        sub main()
+            snoopy = new Dog("Beagle")
+            biplane = new DogHouse(snoopy)
+            print snoopy.kind ' Dog
+            print snoopy.breed ' Beagle
+            print snoopy.isHungry ' true
+            feedAnimal(biplane.puppy)
+            print snoopy.isHungry ' false
+        end sub
+
+        sub feedAnimal(beast as Animal)
+            beast.eat(100)
+        end sub
+        `;
+
+        it('correctly parses the file', () => {
+            program.setFile('source/animal.bs', animalClassCode);
+            program.validate();
+            expect(program.getDiagnostics().length).to.equal(0);
+        });
+
+        it('gets the correct text for m', () => {
+            let animalCode = program.setFile('source/animal.bs', animalClassCode);
+            program.validate();
+            let hover = animalCode.getHover(Position.create(6, 17));
+            expect(hover).to.exist;
+            expect(hover.contents).to.equal('m as Animal');
+
+            hover = animalCode.getHover(Position.create(26, 17));
+            expect(hover).to.exist;
+            expect(hover.contents).to.equal('m as Dog');
+        });
+
+        it('gets the correct text for m.field', () => {
+            let animalCode = program.setFile('source/animal.bs', animalClassCode);
+            program.validate();
+            let hover = animalCode.getHover(Position.create(6, 19));
+            expect(hover).to.exist;
+            expect(hover.contents).to.equal('Animal.kind as string');
```

Comment:
I was just following Typescript's example - it doesn't differentiate between static and not in hovers

Reply:
![image](https://user-images.githubusercontent.com/2544493/124207915-7b262d00-dab4-11eb-8bb3-2efd20a0ea13.png)
![image](https://user-images.githubusercontent.com/2544493/124207929-811c0e00-dab4-11eb-838a-5e8cff396af3.png)

You're right, typescript doesn't differentiate. However, if we do eventually add static properties, we would probably do something like:
```
(static property) Animal.kind
```
So I'm fine with this as-is for now. This type of language feature can be easily changed in the future since it's the presentation layer.